### DISCUSSION TITLE
如何在 Taro3.x 版本， 使用 useReducer 作为全局状态 ？

### DISCUSSION THREAD

**Discussion by zdJOJO** [NervJS/taro#11647] [Q & A]
项目开发中， 遇到了问题。 在使用 taro 开发小程序时， 想使用 useReducer  + useContext 进行跨页面之间的状态共享。

于是，想在 app.ts 文件中注入Provider， 但是 需要将 app.ts 修改为 app.tsx 。

改完之后， 编译直接报错了，  有什么解决方法吗？

`
./src/app.ts
Module build failed (from ./node_modules/babel-loader/lib/index.js):
Error: ENOENT: no such file or directory, open 'D:\public_transit_mini\src\app.ts'
`

**Comment by TheKonka** [comment]
要不 给个 repo 吧？
